.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,80 @@
+name: Benchmark
+# Run on push only for master, if not it will trigger push & pull_request on PRs at the same time
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      # Any update here needs to be done for
+      # - `pull_request` see below
+      # - https://github.com/prisma/prisma/blob/master/src/.buildkite/test/buildkite-entry.sh
+      # - https://github.com/prisma/prisma/blob/master/src/.buildkite/publish/buildkite-entry.sh
+      - "*.md"
+      - ".vscode/**"
+      - "docs/**"
+      - "examples/**"
+      - "LICENSE"
+      - "src/.buildkite/**"
+      - "src/.dockerignore"
+      - "src/scripts/ci/publish.ts"
+      - "src/graphs/**"
+  pull_request:
+    paths-ignore:
+      # Any update here needs to be done for
+      # - `push`see before
+      # - https://github.com/prisma/prisma/blob/master/src/.buildkite/test/buildkite-entry.sh
+      # - https://github.com/prisma/prisma/blob/master/src/.buildkite/publish/buildkite-entry.sh
+      - "*.md"
+      - ".vscode/**"
+      - "docs/**"
+      - "examples/**"
+      - "LICENSE"
+      - "src/.buildkite/**"
+      - "src/.dockerignore"
+      - "src/scripts/ci/publish.ts"
+      - "src/graphs/**"
+jobs:
+  benchmark:
+    name: all
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.pnpm-store
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12"
+
+      - run: bash .github/workflows/setup.sh
+        env:
+          CI: true
+          SKIP_GIT: true
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+      #  This is required as setup.sh can modify pnpm-lock.yml
+      - run: rm -f src/pnpm-lock.yaml
+      - name: Run benchmarks
+        run: yarn bench
+      - name: Store benchmark result
+        uses: rhysd/github-action-benchmark@v1
+        with:
+          name: Benchmark.js Benchmark
+          tool: "benchmarkjs"
+          output-file-path: output.txt
```

Comment:
So where is the output file stored? Looks like it's on git?

Reply:
So on the gh-pages branch?

Reply:
I'm confused that the .cache step is missing?

```
      # Download previous benchmark result from cache (if exists)
      - name: Download previous benchmark data
        uses: actions/cache@v1
        with:
          path: ./cache
          key: ${{ runner.os }}-benchmark
      # Run `github-action-benchmark` action

The step which runs github-action-benchmark does followings:

    Extract benchmark result from the output in output.txt
    Update the downloaded cache file with the extracted result
    Compare the result with the previous result. If it gets worse than previous exceeding 200% threshold, the workflow fails and the failure is notified to you
```

Reply:
output.txt is just used as a cache file on the action. It then parsed then committed to the gh-pages branch ([here](https://github.com/prisma/prisma/blob/gh-pages/dev/bench/data.js#L1)).